main.tf

```
@@ -12,6 +12,8 @@ module "spacelift_config" {
   source  = "cloudposse/stack-config/yaml//modules/spacelift"
   version = "0.22.0"
 
+  count = var.spacelift_stacks_override == {} ? 1 : 0
```

```suggestion
  count = length(var.spacelift_stacks_override) == 0 ? 1 : 0
```

I don't think TF allows comparing maps using `==` operator.
It sems like it works in some cases, but it does not and will lead to unexpected results